src/sgp/SGP.cc

```diff
@@ -542,9 +572,53 @@ int main(int argc, char* argv[])
 	delete cm;
 	GCM = NULL;
 
+#ifdef __ANDROID__
```

Comment:
Probably all other platforms could use a try-catch in the main function

Reply:
I changed it to apply to all platforms.